Sources/Controllers/Map/Layers/OAWeatherRasterLayer.mm

```diff
@@ -159,9 +161,18 @@ - (BOOL) updateLayer
                 layer = OsmAnd::WeatherLayer::Low;
                 break;
         }
-        if (!_provider || wasBandsChanged)
+        
+        // TODO: delete when android team fix alpha changing bug https://github.com/osmandapp/OsmAnd/issues/20533
+        if (_wasAlphaChanged)
+        {
+            _timePeriodEnd = _timePeriodStart;
+        }
+        
+        if (!_provider || wasBandsChanged || _wasAlphaChanged)
         {
             _requireTimePeriodChange = NO;
+            _wasAlphaChanged = NO;
```

Comment:
If provider exists, you need to call `[self.mapView setMapLayerConfiguration` to change alpha instead of recreating provider.